scan/caching_data/bootstrap_nodes.py

```diff
@@ -0,0 +1,89 @@
+import os, json, socket, requests
+#from burst.api.brs.p2p import P2PApi
+from concurrent.futures import ThreadPoolExecutor
+from config.settings import (
+    SIGNUM_NODE,
+    BRS_BOOTSTRAP_NETWORK,
+    BRS_BOOTSTRAP_PEERS,
+    DEFAULT_P2P_PORT,
+)
+from dotenv import (
+    load_dotenv,
+    set_key,
+    find_dotenv,
+)
+from urllib.parse import urlparse
+
+class CachingBootstrapNodes:
+    def __init__(self):
+        # Use env vars (if django restarts for any reason it will re-bootstrap until manually rebooted)
+        load_dotenv(override=True)
+        self.auto_bootstrap_peers = os.environ.get("AUTO_BOOTSTRAP_PEERS", "False").lower() in ("true", "1", "on")
+        self.brs_full_bootstrap = os.environ.get("BRS_FULL_BOOTSTRAP", "true").lower() in ("true", "1", "on")
+        self.bootstrap_peers = self.get_bootstrap_peers()
+        if self.auto_bootstrap_peers and self.brs_full_bootstrap:
+            self.node_bootstrap()
+
+    def get_bootstrap_peers(self):
+        load_dotenv(override=True)        
+        return json.loads(os.environ.get("BRS_BOOTSTRAP_PEERS", "[]"))
+
+    def set_bootstrap_peers(self, peers):
+        set_key(dotenv_path=find_dotenv(), key_to_set="BRS_BOOTSTRAP_PEERS", value_to_set=(json.dumps(peers)), export=False, quote_mode="never")
+
+    def explore_peer(self, peer, address, bootstrap):
+        try:
+            url = (
+                urlparse(
+                    requests.get(f"{address}/burst?requestType=getPeer&peer={peer}").json()['announcedAddress']
+                ) #.scheme # apparently node returns address in scheme instead of http/https
+            )
+            if not url: raise Exception("No URL Found") # Debug info only
+            if int(url.path) == DEFAULT_P2P_PORT: announced = url.scheme
+            elif url.path: announced = f"{url.scheme}:{url.path}"
+            else: raise Exception("No Announced Address")
+            #if not announced: raise Exception("No Announced Address") # Debug info only
+        except: 
+            return
+        else:
+            if announced.endswith((BRS_BOOTSTRAP_NETWORK)):
+                bootstrap.add(announced)
+            return
+
+    ## VERY SLOW, AND HARD ON NODE ##
+    # ONLY RUN ONCE AT FIRST STARTUP #
+    def node_bootstrap(self):
+        print("Running First Time Node Bootstrap...\nThis will take a while and is a blocking function.\nIf this is not needed restart the node with BRS_FULL_BOOTSTRAP=False")
+        """
+            Requests doesn't have docker/container lookup  
+            use urlparse to get the node url(hostname)  
+            followed by socket to get the IP address  
+            then find the bootstrap network peers to use
+        """
+
+        try:
+            local_node = urlparse(SIGNUM_NODE)
+            host = socket.gethostbyname(f'{local_node.hostname}')
```

Comment:
In my .env `SIGNUM_NODE = "127.0.0.1"`
There is always an exception in this line.
`Failed to get bootstrap peers from node: [Errno -2] Name or service not known`

Reply:
Together with the explorer, its own local node is used, because api requests to it occur faster and for this node the parameter "DB.trimDerivedTables=false" in the settings is needed.